feat(schema system): support MySQL no SELECT all rule

rebelice opened the pull request:
## Description

Support MySQL no SELECT all rule.

## Change points

- Implement a new advisor `NoSelectAllAdvisor`
- Add a unit test for `NoSelectAllAdvisor`

Close BYT-406

ecmadao reviewed (approved):
LGTM

tisonkun commented:
I guess BYT-406 is an internal JIRA ticket. Is it possible to expose these subtasks for support schema review so that I may take some of them? I'm interested in this topic but it's unclear which schema rules bytebase wants to support.

dragonly commented:
@tisonkun It's a [Linear Github Magic Word](https://linear.app/docs/github?tabs=206cad22125a#magic-words) and Linear will automatically link it to the corresponding Linear issue.

The magic word should have no visible meaning to the PR itself, and the PR description should be self-contained even without the magic word.

dragonly commented:
And for the schema review policy tasks, FYI @rebelice @d-bytebase

rebelice commented:
@tisonkun Thanks for reminding us. We are currently missing the part of the `schema review `development plan and related discussions. I will create a tracking issue for `schema review `as soon as possible, including the development plan and contributor guidelines. I expect to finish it in two days.

rebelice commented:
@tisonkun Sorry for being much later than expected. I actually wrote the framework for [tracing issues](https://github.com/bytebase/bytebase/issues/1533) last week, but I found the framework code to be messy, so I refactored the relevant code and added the framework for pg advisor and the first example. Now I think it's more readable. Hope this can help you.
